File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.py
vowel = ['a', 'e', 'i', 'o', 'u']

while True:
    password = input()
    if password == 'end':
        break
    
    total_vowel = 0
    vowel_count = 0
    consonant_count = 0
    acceptable = True

    for word in password:
        if word in vowel:
            vowel_count += 1
            consonant_count = 0
            total_vowel += 1
        else:
            consonant_count += 1
            vowel_count = 0
        
        if consonant_count == 3 or vowel_count == 3:
            acceptable = False
            break
    
    if total_vowel == 0:
        acceptable = False

    for index in range(1, len(password)):
        if password[index] == password[index - 1] and password[index] not in ['e', 'o']:
            acceptable = False
            break

    if acceptable:
        print(f"<{password}> is acceptable.")
    else:
        print(f"<{password}> is not acceptable.")

File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        while (true){
            String password = br.readLine();

            if (password.equals("end")) break;

            if (isAcceptable(password)){
                System.out.println("<" + password+"> is acceptable.");
            }else {
                System.out.println("<" + password+"> is not acceptable.");
            }
        }
        br.close();
    }

    public static boolean isAcceptable(String password){
        boolean hasVowel = false;
        int vowelCount = 0;
        int consonantCount = 0;

        for (int i = 0; i < password.length(); i++) {
            char current = password.charAt(i);
            if (isVowel(current)){
                hasVowel = true;
                vowelCount ++;
                consonantCount = 0;
            }else {
                consonantCount ++;
                vowelCount = 0;
            }

            if (vowelCount >= 3 || consonantCount >=3)return false;

            if (i>0 && current == password.charAt(i-1) && current != 'e' && current !='o')
                return false;
        }
        return hasVowel;
    }

    public static boolean isVowel(char c) {
        String[] vowels = {"a", "e", "i", "o", "u"};
        for (String vowel : vowels) {
            if (vowel.charAt(0) == c) return true;
        }
        return false;
    }
}